homeassistant/components/climate/systemair_savecair.py

```diff
@@ -0,0 +1,249 @@
+"""
+SystemAIR platform for VTR 300 Ventilation Unit.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/savecair/
+"""
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    PLATFORM_SCHEMA, SUPPORT_FAN_MODE, SUPPORT_ON_OFF, SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_DEVICE, CONF_NAME, CONF_PASSWORD,
+    CONF_SCAN_INTERVAL, TEMP_CELSIUS, TEMP_FAHRENHEIT)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['websockets==6.0', 'python-systemair-savecair==0.0.2']
+
+ATTR_SUPPLY_AIR_TEMPERATURE = "supply_air_temperature"
+ATTR_SUPPLY_AIR_FAN_SPEED = "supply_air_fan_speed"
+ATTR_EXTRACT_AIR_FAN_SPEED = "extract_air_fan_speed"
+
+CONF_IAM_ID = "iam_id"
+
+DEFAULT_NAME = "SystemAIR"
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_DEVICE): cv.string,
+    vol.Required(CONF_IAM_ID): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the SystemAIR platform."""
+    name = config.get(CONF_NAME)
+    device = config.get(CONF_DEVICE)
+    iam_id = config.get(CONF_IAM_ID)
+    password = config.get(CONF_PASSWORD)
+
+    climate = SystemAIRClimate(name, device, iam_id, password)
+    async_add_entities([climate])
+
+
+class SystemAIRClimate(ClimateDevice):
+    """Representation of the climate sensor."""
+
+    def __init__(self, name, unit, iam_id, password):
+        """Construct the SystemAIR Climate Device."""
+        from systemair.savecair.client import SavecairClient
+        self._name = name
+
+        """Initialize the climate device."""
+        self._support_flags = SUPPORT_TARGET_TEMPERATURE
+
+        if unit == "vtr_300":
+            self._support_flags = self._support_flags | SUPPORT_FAN_MODE
+            self._support_flags = self._support_flags | SUPPORT_OPERATION_MODE
+            self._support_flags = self._support_flags | SUPPORT_ON_OFF
+
+        # Create savecair client
+        self._client = SavecairClient(iam_id, password)
+        self._client.update_cb.append(self.update_callback)
```

Comment:
Please move callback registration to entity coroutine method `async_added_to_hass`. Otherwise if an update comes in before the entity has been added to home assistant there will be an error.